src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Eggress/AzureBlob/AzureBlobEgressProviderTests.cs

```diff
@@ -0,0 +1,295 @@
+            Uri sasUri = containerClient.GenerateSasUri(sasBuilder);
+
+            AzureBlobEgressProviderOptions providerOptions = ConstructEgressProviderSettings(containerClient, sasToken: sasUri.Query);
+            EgressArtifactSettings artifactSettings = ConstructArtifactSettings();
+
+            // Act & Assert
+            await Assert.ThrowsAsync<EgressException>(async () => await egressProvider.EgressAsync(providerOptions, UploadAction, artifactSettings, CancellationToken.None));
+        }
+
+        private Task<Stream> UploadAction(CancellationToken token)
+        {
+            return Task.FromResult<Stream>(new FileStream(_testUploadFile, FileMode.Open, FileAccess.Read));
+        }
+
+        private async Task<BlobContainerClient> ConstructBlobContainerClientAsync(string containerName = null, bool create = true)
+        {
+            BlobServiceClient serviceClient = new(_azuriteFixture.Account.ConnectionString);
+
+            containerName ??= Guid.NewGuid().ToString("D");
+            BlobContainerClient containerClient = serviceClient.GetBlobContainerClient(containerName);
+
+            if (create)
+            {
+                await containerClient.CreateIfNotExistsAsync();
+            }
+
+            return containerClient;
+        }
+
+        private async Task<QueueClient> ConstructQueueContainerClientAsync(string queueName = null, bool create = true)
+        {
+            QueueServiceClient serviceClient = new(_azuriteFixture.Account.ConnectionString);
+
+            queueName ??= Guid.NewGuid().ToString("D");
+            QueueClient queueClient = serviceClient.GetQueueClient(queueName);
+
+            if (create)
+            {
+                await queueClient.CreateIfNotExistsAsync();
+            }
+
+            return queueClient;
+        }
+
+        private EgressArtifactSettings ConstructArtifactSettings(int numberOfMetadataEntries = 2)
+        {
+            EgressArtifactSettings settings = new()
+            {
+                ContentType = ContentTypes.ApplicationOctetStream,
+                Name = Guid.NewGuid().ToString("D")
+            };
+
+            for (int i = 0; i < numberOfMetadataEntries; i++)
+            {
+                settings.Metadata.Add($"key_{i}", Guid.NewGuid().ToString("D"));
+            }
+
+            return settings;
+        }
+
+        private AzureBlobEgressProviderOptions ConstructEgressProviderSettings(BlobContainerClient containerClient, QueueClient queueClient = null, string sasToken = null)
+        {
+            AzureBlobEgressProviderOptions options = new()
+            {
+                AccountUri = new Uri(_azuriteFixture.Account.BlobEndpoint),
+                ContainerName = containerClient.Name,
+                QueueAccountUri = (queueClient == null) ? null : new Uri(_azuriteFixture.Account.QueueEndpoint),
+                QueueName = queueClient?.Name,
+                BlobPrefix = Guid.NewGuid().ToString("D")
```

Comment:
We should have a test that doesn't provide a prefix.

Reply:
It's fine if this is a follow up item. There are some other cases that we'll want to consider covering it time since we've made specific effort to support scenarios, such as partial blob uploads.